java-client/src/generated/java/org/opensearch/client/opensearch/snapshot/SnapshotInfo.java

```diff
@@ -50,55 +57,64 @@
 import org.opensearch.client.util.ObjectBuilder;
 import org.opensearch.client.util.ObjectBuilderBase;
 
-// typedef: snapshot._types.SnapshotInfo
+// typedef: snapshot.SnapshotInfo
 
 @JsonpDeserializable
+@Generated("org.opensearch.client.codegen.CodeGenerator")
 public class SnapshotInfo implements PlainJsonSerializable {
+
+    @Nonnull
     private final List<String> dataStreams;
 
     @Nullable
     private final Time duration;
 
     @Nullable
-    private final String durationInMillis;
+    private final Long durationInMillis;
 
     @Nullable
-    private final Time endTime;
+    private final String endTime;
 
     @Nullable
-    private final String endTimeInMillis;
+    private final Long endTimeInMillis;
 
+    @Nonnull
     private final List<SnapshotShardFailure> failures;
 
     @Nullable
     private final Boolean includeGlobalState;
 
+    @Nonnull
     private final List<String> indices;
 
-    private final Map<String, IndexDetails> indexDetails;
-
+    @Nonnull
     private final Map<String, JsonData> metadata;
 
     @Nullable
-    private final String reason;
+    private final Long pinnedTimestamp;
 
     @Nullable
-    private final String repository;
+    private final String reason;
 
-    private final String snapshot;
+    @Nullable
+    private final Boolean remoteStoreIndexShallowCopy;
 
     @Nullable
     private final ShardStatistics shards;
 
+    @Nonnull
+    private final String snapshot;
+
     @Nullable
-    private final Time startTime;
+    private final String startTime;
```

Comment:
Shouldn't the `startTime` stayed as `Time`? Or DateTime?

Reply:
It seems that common___DateTime is being converted to a String.
Should I override them individually in the Overrides class?
```
            .with(
                schema("snapshot._common", "SnapshotInfo"),
                so -> so.withProperties(
                    mb -> mb.with("start_time", po -> po.withMappedType(Types.Client.OpenSearch._Types.Time))
                        .with("end_time", po -> po.withMappedType(Types.Client.OpenSearch._Types.Time))
                )
            )
```

Reply:
So the [`Time`](https://github.com/opensearch-project/opensearch-java/blob/main/java-client/src/main/java/org/opensearch/client/opensearch/_types/Time.java#L51) type is (confusingly) actually the Duration type, which is why it's used for `clusterManagerTimeout` etc. So `startTime` being `Time` was incorrect anyway. 

I hadn't added any special handling for `DateTime` as the only property I'd run across so far in my generation with that schema is [`OpenSearchVersionInfo.buildDate`](https://github.com/opensearch-project/opensearch-java/blob/dc41a99f1fd76fa59db8aa28c5466602cbcc6eb1/java-client/src/main/java/org/opensearch/client/opensearch/_types/OpenSearchVersionInfo.java#L49) which was already a String type.

Reply:
I see, thanks

Reply:
Happy for us to add a special type mapping for DateTime, it's only `OpenSearchVersionInfo.buildDate` is a bit special/annoying because it can return the string `"unknown"` so it needs to be String where others can be whatever date type.

Reply:
Sounds great to me, thanks!